src/test/java/bruno/task/DeadlineTest.java

```diff
@@ -0,0 +1,43 @@
+package bruno.task;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.time.format.DateTimeParseException;
+
+import org.junit.jupiter.api.Test;
+
+public class DeadlineTest {
+
+    @Test
+    void testConstructor_incorrectDateFormat_exceptionThrown() {
+        assertThrows(DateTimeParseException.class, ()
+                -> new Deadline("sign up for competition", "01-09-2023 5pm"));
+    }
+
+    @Test
+    void testGetString() {
```

Comment:
Perhaps can specify what is the input string, since this is a special case of date string